bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -536,6 +536,62 @@ mod tests {
         Ok(())
     }
 
+    /// `trust_location()` calls `s2n_config_set_verification_ca_location()`, which has the legacy behavior
+    /// of enabling OCSP on clients. Since we do not want to replicate that behavior in the Rust bindings,
+    /// this test verifies that `trust_location()` does not turn on OCSP. It also verifies that turning
+    /// on OCSP explicitly still works when `trust_location()` is called.
+    #[test]
+    fn trust_location_does_not_change_ocsp_status() -> Result<(), Error> {
+        let pem_dir = Path::new(concat!(env!("CARGO_MANIFEST_DIR"), "/../../../tests/pems"));
+        let mut cert = pem_dir.to_path_buf();
+        cert.push("rsa_4096_sha512_client_cert.pem");
+        let mut key = pem_dir.to_path_buf();
+        key.push("rsa_4096_sha512_client_key.pem");
+
+        const OCSP_IANA_EXTENSION_ID: u16 = 5;
+
+        for enable_ocsp in [true, false] {
+            let config = {
+                let mut config = crate::config::Builder::new();
+
+                if enable_ocsp {
+                    config.enable_ocsp()?;
+                }
+
+                config.set_security_policy(&security::DEFAULT_TLS13)?;
+                config.set_verify_host_callback(InsecureAcceptAllCertificatesHandler {})?;
+                config.set_client_hello_callback(HasExtensionClientHelloHandler {
+                    // This client hello handler will check for the OCSP extension
+                    extension_iana: OCSP_IANA_EXTENSION_ID,
+                    extension_expected: enable_ocsp,
+                })?;
+                config.load_pem(&fs::read(&cert)?, &fs::read(&key)?)?;
+                config.trust_location(Some(&cert), None)?;
+                config.build()?
+            };
+
+            let server = {
```

Comment:
nit: `tls_pair()` creates a client/server pair similar to this and might simplify the test, is there a reason why this function wasn't used?

Reply:
I needed to set the waker on the server. I've added a `connection_mut` method to the `Harness` now so I can use `tls_pair()` while also modifying the server.